include/swift/AST/NameLookup.h

```diff
@@ -110,13 +150,13 @@ class UnqualifiedLookup {
   /// is used to determine which declarations in that body are visible.
   UnqualifiedLookup(DeclName Name, DeclContext *DC, LazyResolver *TypeResolver,
                     SourceLoc Loc = SourceLoc(), Options options = Options());
-
-  SmallVector<LookupResultEntry, 4> Results;
+  using ResultsVector = SmallVector<LookupResultEntry, 4>;
```

Comment:
Are you using this type outside of the class? If not, why create this? If you are using it outside, it is better to use SmallVectorImpl.

Reply:
Yes,` UnqualifedLookupFactory` uses it, and no, the Impl type won't do. But is there a better way?

Reply:
Why wont it do. If you are passing it to someone, you can use SmallVectorImpl< LookupResultEntry> &

Reply:
In `unqualifiedLookup.cpp`:
```
  if (compareToASTScopes && useASTScopesForExperimentalLookupIfEnabled()) {
    ResultsVector results;
    size_t indexOfFirstOuterResult = 0;
    UnqualifiedLookupFactory scopeLookup(Name, DC, TypeResolver, Loc, options,
                                         results, indexOfFirstOuterResult);
```
The code needs to create a results vector, so it can make another factory for comparison purposes. Publishing the type seemed like the clearest way. I suppose the code could use a `decltype` instead but it's not clearer.